### PR TITLE
Release v3.7.0

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -125,6 +125,8 @@ As you can see, the edge between `develop` and `call-ws` changed to grey &mdash;
 You can easily rename a branch directly from Git Machete.
 Open a context menu for a branch and choose `Rename...` action.
 
+For the tracked branches, it is also possible to rename the remote branch.
```
Comment (mkondratek) [inline docs/features.md:128]: @PawelLipski , can you pls piggyback a fix here? To reflect [this](https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1497#discussion_r1127556124) better (currently it is very inaccurate). sorry I have not fixed it in the first place

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RenameBackgroundable.java [hunk 2]
```diff
@@ -73,6 +79,73 @@ public void doRun(ProgressIndicator indicator) {
...
+      // and the remote branch with the new name exist.
+      return;
+    }
+
+    if (!setUpstreamErrorOutput.isEmpty() && setUpstreamErrorOutput.get(0)
+        .equals("fatal: the requested upstream branch '${renamedRemoteName}' does not exist")) {
+      val unsetUpstream = unsetUpstream(newBranchName);
+      val unsetUpstreamErrorOutput = unsetUpstream.getErrorOutput();
+      if (unsetUpstream.success()) {
+        // the requested upstream branch 'origin/<new-branch-name>' does not exist
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RenameBackgroundable.java:111]: hm, we want to keep that output, right?

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRenameAction.java [hunk 3]
```diff
@@ -81,13 +82,14 @@ public void actionPerformed(AnActionEvent anActionEvent) {
...
+    val remotes = new ArrayList<>(gitRepository.getRemotes());
 
-    val gitNewBranchDialog = new GitNewBranchDialog(project, Collections.singletonList(gitRepository),
+    val gitNewBranchDialog = new MyGitNewBranchDialog(project, Collections.singletonList(gitRepository),
         getString("action.GitMachete.BaseRenameAction.description").fmt(branch.getName()),
         branch.getName(),
         /* showCheckOutOption */ false,
         /* showResetOption */ false,
-        /* showSetTrackingOption */ false,
+        /* showSetTrackingOption */ branch.getRemoteTrackingBranch() != null,
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRenameAction.java:92]: nit: it is not `showSetTrackingOption`. it's `showKeepRemoteOption`

**Review by mkondratek — COMMENTED** [review]
I post-self-reviewed here XD

File: docs/features.md (hunk 1 above)
Comment (PawelLipski) [reply docs/features.md:128]: #1611

**Pull request merged**